src/main/java/com/coherentsolutions/training/aqa/java/web/nikonova/browsers/BrowsersSettings.java

```diff
@@ -0,0 +1,102 @@
+
+    public BrowsersSettings(String environment, String browser) {
+        this.environment = environment;
+        this.browser = browser;
+    }
+
+    public static void driverTeardown() {
+        getDriver().quit();
+        driver.remove();
+    }
+
+    public static WebDriver getDriver() {
+        return driver.get();
+    }
+
+    public void setBrowserParameters() throws Exception {
+        if (environment.equals(BrowsersProperties.getBrowserProperties("env.local"))) {
+            if (browser.equals(BrowsersProperties.getBrowserProperties("browser.chrome"))) {
+                driver.set(ThreadGuard.protect(new ChromeDriver()));
+            } else if (browser.equals(BrowsersProperties.getBrowserProperties("browser.firefox"))) {
+                driver.set(ThreadGuard.protect(new FirefoxDriver()));
+            } else {
+                throw new Exception("Browser is unavailable");
+            }
+        } else if (environment.equals(BrowsersProperties.getBrowserProperties("env.grid"))) {
+            if (browser.equals(BrowsersProperties.getBrowserProperties("browser.chrome"))) {
+                chromeOptions = new ChromeOptions();
+                chromeOptions.setPlatformName(BrowsersProperties.getBrowserProperties(String.format("%s.chrome.browser.platform", environment)));
+                chromeOptions.setBrowserVersion(BrowsersProperties.getBrowserProperties(String.format("%s.chrome.browser.version", environment)));
+                driver.set(ThreadGuard.protect(new RemoteWebDriver(new URL(BrowsersProperties.getBrowserProperties("url.grid")), chromeOptions)));
+            } else if (browser.equals(BrowsersProperties.getBrowserProperties("browser.firefox"))) {
+                firefoxOptions = new FirefoxOptions();
+                firefoxOptions.setPlatformName(BrowsersProperties.getBrowserProperties(String.format("%s.firefox.browser.platform", environment)));
+                firefoxOptions.setBrowserVersion(BrowsersProperties.getBrowserProperties(String.format("%s.firefox.browser.version", environment)));
+                driver.set(ThreadGuard.protect(new RemoteWebDriver(new URL(BrowsersProperties.getBrowserProperties("url.grid")), firefoxOptions)));
+            } else {
+                throw new Exception("Browser is unavailable");
+            }
+        } else if (environment.equals(BrowsersProperties.getBrowserProperties("env.sauceLabs"))) {
+            if (browser.equals(BrowsersProperties.getBrowserProperties("browser.chrome"))) {
+                driver.set(ThreadGuard.protect(sauceLabsTestOnChrome()));
+            } else if (browser.equals(BrowsersProperties.getBrowserProperties("browser.firefox"))) {
+                driver.set(ThreadGuard.protect(sauceLabsTestOnFirefox()));
+            } else {
+                throw new Exception("Browser is unavailable");
+            }
+        } else {
+            throw new Exception("Environment is unavailable");
+        }
+    }
```

Comment:
too complicated method. I suggest to use factory pattern here (DriverFactory or something like this)